modules/gdscript/gdscript_editor.cpp

```diff
@@ -2005,7 +2005,7 @@ static bool _guess_identifier_type(GDScriptParser::CompletionContext &p_context,
 	}
 
 	// Check autoloads.
-	if (ProjectSettings::get_singleton()->has_autoload(p_identifier)) {
+	if (ProjectSettings::get_singleton()->has_autoload(p_identifier) && GDScriptLanguage::get_singleton()->get_named_globals_map().has(p_identifier)) {
```

Comment:
I suggest a comment explaining why `get_named_globals_map()` may not contain `p_identifier`.

```suggestion
	// Check autoloads. Identifier may not be in the globals map if "Global Variable" is disabled.
	if (ProjectSettings::get_singleton()->has_autoload(p_identifier) && GDScriptLanguage::get_singleton()->get_named_globals_map().has(p_identifier)) {
```

Note: Do not apply my changes through GitHub, as I could not replace the previous comment.